### PR TITLE
Add scripts to access API

### PR DESCRIPTION
**Pull request by abhidg**
Draft version, comments welcome! I haven't put up versioned dependencies as the libraries are fairly stable, and people might be stuck on older versions; also not sure what the standard is in R.﻿

### CODE REVIEW

**Review by iamleeg — APPROVED** [review]
Looks good, bear in mind I don't understand R very much! Here's the next challenge: is there a way to build an integration test to ensure we don't regress the API?

**Review by jim-sheldon — APPROVED** [review]

**Comment by jim-sheldon** [comment]
I also can't offer much around R...yet :)
Once the end-to-end testing environment is done I can give a demo and show how to add new tests.

If it's helpful I put a couple assert-like functions in [this file](https://github.com/globaldothealth/list/blob/main/dev/run_session.py) to do things like check the file for valid structure and contents.

File: api/python/gdh.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import os
+import io
+import json
+import hashlib
+
+import requests
+import pandas as pd
+
+GH = "https://data.covid-19.global.health"
```
Comment (jim-sheldon) [inline api/python/gdh.py:22]: Could this be an environment variable, with the current value as the default?
That way we could reuse it for other environments easily.

File: api/python/gdh.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+downloadAsync = "/api/cases/downloadAsync"
+
+def cases(query, apikey, server=GH):
+    res = requests.post(
+        f"{server}{downloadAsync}",
+        data=json.dumps({"format": "csv", "query": query}),
```
Comment (jim-sheldon) [inline api/python/gdh.py:28]: "csv" might be better as a default `kwarg` value to extend this function to other formats

File: api/python/gdh.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        headers={"Content-Type": "application/json", "X-API-Key": apikey},
+    )
+    if res.status_code != 200:
+        print(res.text)
+        return False
```
Comment (jim-sheldon) [inline api/python/gdh.py:33]: I think a `raise` might be better here, so the return value of the function is of a single type

File: api/python/gdh.py (hunk 2 above)
Comment (abhidg) [reply api/python/gdh.py:70]: `cases()` is about providing the case data as a dataframe without regard to the underlying format, I'll add a function to download in the specific format requested separately.

File: api/python/gdh.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    "admin1",
+    "admin2",
+    "admin3",
+    "variant",
+    "dateconfirmedafter",
+    "dateconfirmedbefore",
+}
+
+
+def qs(**kwargs):
```
Comment (jim-sheldon) [inline api/python/gdh.py:49]: Does this function turn filters into a string? If so, maybe a better name would be something like `stringify_filters`?

File: api/python/gdh.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+def cases_cachefile(query, server=GDH_URL, folder="cache"):
+    to_hash = f"{query.strip()}|{server}"
+    sha256 = hashlib.sha256(to_hash.encode("utf-8")).hexdigest()
+    return f"{folder}/{sha256}.csv"
+
+
+# cases_cached returns a cached copy of cases if it exists,
+# otherwise saves to cache
+def cases_cached(query, apikey, server=GDH_URL, refresh=False, folder="cache"):
```
Comment (jim-sheldon) [inline api/python/gdh.py:92]: In general I find functions named as verbs and objects named as nouns to be a better pattern to follow...i.e. a function named something like `get_cached_cases` will do just that.

File: .github/workflows/api-python-tests.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    paths:
+    - '.github/workflows/api-python-tests.yml'
+    - 'api/python/**'
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: data-serving/scripts/prune-uploads
```
Comment (iamleeg) [inline .github/workflows/api-python-tests.yml:19]: Surely a mistake?

**Pull request merged**